src/btree/bt_curnext.c

```diff
@@ -692,8 +702,9 @@ __wt_btcur_next(WT_CURSOR_BTREE *cbt, bool truncating)
                 total_skipped += skipped;
                 break;
             case WT_PAGE_ROW_LEAF:
-                ret = __cursor_row_next(cbt, newpage, restart, &skipped);
-                total_skipped += skipped;
+                ret = __cursor_row_next(cbt, newpage, restart, &skipped, prefix);
+                if (ret == WT_NOTFOUND && F_ISSET(&cbt->iface, WT_CURSTD_PREFIX_SEARCH))
+                    return (WT_NOTFOUND);
```

Comment:
I'd prefer to have this handled in the `if (ret != WT_NOTFOUND)` condition after the case statement. Adding new return paths out of functions can have odd behavior.

Reply:
This is interesting, I tried that originally but for reasons that I haven't had time to determine if we exit the normal way from this function we get about 50x slower. So I'm thinking theres an edge case here that I haven't considered.

All functions `__wt_btcur_search_near`, `__wt_btcur_next`, `__wt_btcur_prev` reset the cursor on returning WT_NOTFOUND so if the `__wt_btcur_next` traversal resets the cursor then the `__wt_btcur_prev` would start from the wrong place in the tree I believe in this loop in `__wt_btcur_search_near`:[loop]( https://github.com/wiredtiger/wiredtiger/blob/3227f141a3e94a44570fc479250f8d1a7de28b32/src/btree/bt_cursor.c#L706) But that would imply the original implementation of search near is incorrect.